yocto-base/00_flash_and_verify.robot

```diff
@@ -0,0 +1,49 @@
+*** Settings ***
+Library             Collections
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+Resource            ../../sonoff-rest-api/sonoff-api.robot
+Resource            ../../rtectrl-rest-api/rtectrl.robot
+Resource            ../../variables.robot
+Resource            ../../keywords.robot
+Resource            ../lib/sd-wire.robot
+
+Suite Setup         Prepare Test Suite
+Suite Teardown      Log Out And Close Connection
+
+
+*** Test Cases ***
+Verify number of connected SD Wire devices
+    [Documentation]    Test confirms the number of connected SD Wire devices to
+    ...    the DUT platform.
+    ${sd_wire_id_list}=    Get List Of SD Wire Ids
+    ${length_of_sd_wire_id_list}=    Get Length    ${sd_wire_id_list}
+    # this is done to check whether the id of the SD Wire that's connected
+    # to the RTE matches the expected value.
+    Should Be Equal    ${length_of_sd_wire_id_list}    ${SD_WIRES_CONNECTED}
+    FOR    ${index}    IN RANGE    0    ${SD_WIRES_CONNECTED}
+        Should Be Equal    ${sd_wire_id_list[${index}]}    ${SD_WIRE_SERIAL1}
+    END
+
+Flash platform and verify
+    [Documentation]    This test flashes the DUT connected to the RTE through
+    ...    the SD Wire, then attempts to log into it over serial to see whether
+    ...    it works.
+    IF    '${PREV_TEST_STATUS}' == 'FAIL'
+        Fail    'Incorrect number of connected SD Wire devices.'
+    END
+    Variable Should Exist    ${DUT_PASSWORD}
+    # flashing
+    Flash SD Card Via SD Wire    ${FILE_BMAP}    ${FILE_GZ}    ${SD_WIRE_SERIAL1}
+    # telnet
+    Telnet.Set Prompt    :~#
+    Telnet.Read Until    login:
+    Telnet.Write Bare    \n
+    Telnet.Login    root    ${DUT_PASSWORD}
+    ${output}=    Telnet.Execute Command    sh -c "cat /etc/os-release | grep VERSION_ID"
```

Comment:
If you already put complex shell expression in, you may just as well put one more and save two lines of robot thanks to that:

```
cat /etc/os-release | grep VERSION_ID | cut -d '=' -f 2
```

Reply:
BTW, checking for version of Linux version sounds like a generic Linux keyword to be defined, so it can be reused.

Reply:
@macpijan adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/22/commits/909324f803fe083d61d8fbfc84429dbb1b2224b5#diff-322efff20647f11c797d3f64ba9852d27c8a23969970ad17b19b7b9e7c7d7029L40-R43). I do not understand the second comment. Are You suggesting that I should add a new keyword to `keywords.robot` that checks the version?

Reply:
Yes. It should be very common activity to do to retreve Linux distro version. So you should create a keyword that just returns Linux version from the OS. 
But we are not adding new kewords in keyword.robot. Please read the guidelines carefully. You should create a new library, e.g. `lib/linux.robot`.

Reply:
@macpijan are you sure? it would be a keyword that only has two lines:
```
    ${output}=    Telnet.Execute Command    sh -c "cat /etc/os-release | grep VERSION_ID | cut -d '=' -f 2"
    ${output}=    Get Line    ${output}    0
    [Return]    ${output}
```
and it wouldn't be super generic (aka wouldnt work over ssh, and there would need to be a separate keyword for ssh). Is a new keyword necessary for this?

Reply:
@WiktorG351 maybe we can determine the correct connection based on the platform-config file? Such as `${DUT_CONNECTION_METHOD}` which we already use in our tests?

Then, based on the value of the variable, you can extend the keyword to handle telnet and ssh command execution, returning the same variable in both cases.

Reply:
@artur-rs okay, adressed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/22/commits/4db9447d6aca6f1c3a625514b42a6cdaa9c34d88)